src/core/subxt_subscription.rs

```diff
@@ -69,6 +73,23 @@ pub enum SubxtDisputeResult {
 	TimedOut,
 }
 
+/// How to subscribe to subxt blocks
+#[derive(strum::Display, Debug, Clone, Copy, clap::ValueEnum)]
+pub enum SubxtSubscriptionMode {
+	/// Subscribe to all blocks
+	All,
```

Comment:
If we choose this do we get notifications twice for a block, when it's first imported and then finalized ? How do the tools cope with this ?

Reply:
I have not tested this I'm afraid. Presumably, it will be broken (unless my next WiP PR about collector-as-a-service is finished). However, without this PR the tool is definitely broken as it uses `subscribe_all` by default.

Reply:
Let’s just not expose the subscribe all here so we don’t break things at all

Reply:
Done.